include/openthread/dataset.h

```diff
@@ -426,6 +426,25 @@ otError otDatasetSendMgmtPendingSet(otInstance *                aInstance,
                                     const uint8_t *             aTlvs,
                                     uint8_t                     aLength);
 
+/**
+ * This function generates PSKc from a given pass-phrase, network name, and extended PAN Id.
+ *
+ * PSKc is used to establish the Commissioner Session.
+ *
+ * @param[in]  aPassPhrase   The commissioning pass-phrase.
+ * @param[in]  aNetworkName  The network name for PSKc computation.
+ * @param[in]  aExtPanId     The extended PAN ID for PSKc computation.
+ * @param[out] aPskc         A pointer to variable to output the generated PSKc.
+ *
+ * @retval OT_ERROR_NONE          Successfully generate PSKc.
+ * @retval OT_ERROR_INVALID_ARGS  If any of the input arguments is invalid.
+ *
+ */
+otError otDatasetGeneratePskc(const char *           aPassPhrase,
+                              const char *           aNetworkName,
```

Comment:
What about using `otNetworkName` defined in this file?

Reply:
otThreadGetNetworkName() returns `const char*`. I think keeping `const char*` is simpler. Thoughts?

Reply:
I would prefer using defined types when available. Makes code better understandable. Specially useful when searching for occurrences of this among the whole project files.